### PR TITLE
Add a gallery with tests and doc based on examples given in #219, #244 and #251

### PR DESCRIPTION
**Pull request by adrien-berchet**
We had several questions about it so I think it can be good to add some doc on how to use TypeDecorator

### CODE REVIEW

**Comment by adrien-berchet** [comment]
@elemoine I wonder if we could not make a gallery in the doc to collect examples of this kind, so when we provide a new example in an issue we can add it to the gallery. I think it would be easier for people to find relevant examples if they can find them in the doc instead of having digging in old issues.

**Comment by elemoine** [comment]
> @elemoine I wonder if we could not make a gallery in the doc to collect examples of this kind, so when we provide a new example in an issue we can add it to the gallery. I think it would be easier for people to find relevant examples if they can find them in the doc instead of having digging in old issues.

Yep, I fully support that idea! Sounds great.

File: doc/orm_tutorial.rst [hunk 1]
```diff
@@ -361,6 +361,19 @@ measurement functions can be applied to
     >>> print '%s: %f' % (lake.name, bufferarea)
     Majeur: 21.485781
 
+Automatically use a function at insert or select
+------------------------------------------------
+
+Sometimes the application always want to apply a function in an insert or in a select.
```
Comment (elemoine) [inline doc/orm_tutorial.rst:367]: Having both "sometimes" and "always" here sounds strange to.

Also s/want/wants

**Comment by adrien-berchet** [comment]
Thanks, I fixed according to your comments.
I also setup the gallery and moved the example to it.

**Comment by adrien-berchet** [comment]
@elemoine I added two more examples so I re-request a review before merging this.
Also, how do you publish the new doc on readthedoc?

File: doc/index.rst [hunk 2]
```diff
@@ -75,6 +75,18 @@ system. If you're new to GeoAlchemy 2 start with this.
...
+
+Gallery
+---------
+
+.. toctree::
+   :hidden:
+
+   gallery/index
+
+The :ref:`gallery` page shows examples of the functionalities of GeoAlchemy 2.
```
Comment (elemoine) [inline doc/index.rst:87]: I'd say "of GeoAlchemy 2's functionality".

File: tests/gallery/test_disable_wrapping.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
+"""
+Disable wrapping in select
+==========================
+
+If the application wants to build queries with GeoAlchemy 2 and get them as strings,
```
Comment (elemoine) [inline tests/gallery/test_disable_wrapping.py:5]: s/get/gets

File: tests/gallery/test_disable_wrapping.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+the wrapping of geometry columns by a `ST_AsEWKB()` function might be annoying. In
```
Comment (elemoine) [inline tests/gallery/test_disable_wrapping.py:6]: s/by/with

**Comment by elemoine** [comment]
A few more comments, but it looks great. Feel free to merge whenever you think it's ready. Thanks a lot.

**Comment by adrien-berchet** [comment]
Ok, I will try to merge all the current PR today so we can release a new version.
Thanks!

**Pull request merged**